boards/PDM/Src/Io/Io_Efuse_22XS4200.c

```diff
@@ -0,0 +1,198 @@
+}
+
+uint16_t
+Io_Efuse_22XS4200_ReadReg(uint8_t register_address, struct E_Fuse *e_fuse)
+{
+    HAL_StatusTypeDef status;
+    uint16_t          command    = 0x0000U;
+    uint8_t           parity_bit = 0;
+    uint16_t          tx_data[]  = { 0x0000U };
+    uint16_t          rx_data[]  = { 0x0000U };
+
+    command =
+            (uint16_t)((SI_STATR_0_ADDR & EFUSE_ADDR_MASK) << EFUSE_ADDR_SHIFT);
+    /* shift bit 3 of the address (SOA3: the channel number) to the D13 position
+     */
+    command = (uint16_t)(
+            command | ((register_address & SOA3_MASK) << EFUSE_ADDR_SHIFT) |
+            (register_address & (SOA2_MASK | SOA1_MASK | SOA0_MASK)));
+    /* invert watchdog bit state */
+    if (e_fuse->watch_dog_state)
+    {
+        command |= WATCH_DOG_BIT;
+    }
+    else
+    {
+        command &= ~(WATCH_DOG_BIT);
+    }
+    /* invert watchdog bit state for next write */
+    e_fuse->watch_dog_state = !e_fuse->watch_dog_state;
+    /* compute parity */
+    parity_bit = Io_Efuse_Cal_Parity_Bit(command);
+    /* add parity value */
+    command    = (uint16_t)command | ((parity_bit & 1U) << PARITY_BIT_SHIFT);
+    tx_data[0] = command;
+    status     = Io_Efuse_22XS4200_ReadFromEfuse(
+            tx_data, rx_data, e_fuse->chip_select.GPIO_Port,
+            e_fuse->chip_select.GPIO_Pin);
+    UNUSED(status);
+    return rx_data[0];
+}
+
+uint8_t Io_Efuse_Cal_Parity_Bit(uint16_t spi_command)
+{
+    uint8_t parity_bit;
+    /* calculate parity for bits in spi command */
+    for (parity_bit = 0; spi_command > 0; spi_command >>= 1)
+    {
+        parity_bit ^= (spi_command & 1);
+    }
+    return parity_bit;
+}
+
+HAL_StatusTypeDef Io_Efuse_22XS4200_WriteToEfuse(
+        uint16_t *    TxData,
+        GPIO_TypeDef *GPIO_Port,
+        uint16_t      GPIO_Pin)
+{
+    HAL_StatusTypeDef status;
+
+    HAL_GPIO_WritePin(GPIO_Port, GPIO_Pin, GPIO_PIN_RESET);
+    status = HAL_SPI_Transmit(efuse_spi_handle, (uint8_t *)TxData, 1U, 100U);
+    while (efuse_spi_handle->State != HAL_SPI_STATE_READY)
+        ;
+    HAL_GPIO_WritePin(GPIO_Port, GPIO_Pin, GPIO_PIN_SET);
+
+    return status;
+}
+
+HAL_StatusTypeDef Io_Efuse_22XS4200_ReadFromEfuse(
+        uint16_t *    TxData,
+        uint16_t *    RxData,
+        GPIO_TypeDef *GPIO_Port,
+        uint16_t      GPIO_Pin)
+{
+    HAL_StatusTypeDef status;
+    // Send command to read from status register
+    // Data is returned on the following SPI transfer
+    HAL_GPIO_WritePin(GPIO_Port, GPIO_Pin, GPIO_PIN_RESET);
+    status = HAL_SPI_TransmitReceive(
+            efuse_spi_handle, (uint8_t *)TxData, (uint8_t *)RxData, 1U, 100U);
+    while (efuse_spi_handle->State != HAL_SPI_STATE_READY)
+        ;
+    HAL_GPIO_WritePin(GPIO_Port, GPIO_Pin, GPIO_PIN_SET);
+    // Receive data from E-fuse (transmitted data is ignored) RxData(MISO) is
+    // correct
+    *TxData = 0xFFFF;
+    HAL_GPIO_WritePin(GPIO_Port, GPIO_Pin, GPIO_PIN_RESET);
+    status &= HAL_SPI_TransmitReceive(
```

Comment:
If transmit is ignored can we use HAL_SPI_Receive ?

Reply:
Ya looks like` HAL_SPI_Receive()`; just calls `HAL_SPI_TransmitReceive()` with dummy Tx values. Switched to `HAL_SPI_Transmit()`; and` HAL_SPI_Receive();`